src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -148,14 +149,32 @@ public void Start()
             _initialInstrumentEnumerationComplete();
         }
 
+        public void Update()
```

Comment:
This is called from within `MetricsEventSource` after adding the new instruments and partially mimics what `Start` does - there's a comment inline, but I wasn't sure if it makes sense to repeatedly call `_initialInstrumentEnumerationComplete` every time, given that it might be a bit of a misnomer (and I'm not sure what assumptions might exist around this).

Reply:
We can frame this in terms the publicly visible behavior of MetricsEventSource: When multiple consumers all listen to this source does each of them get a [InitialEnumerationComplete](https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Metrics/MetricsEventSource.cs,157) or does only the first session get that event?

I think the behavior we'll find more useful is that all sessions get the event at least once, otherwise it is problematic for new sessions to discover what instruments exist. Shared sessions will need to be prepared to ignore duplicate events they observe later.

Likewise I'd apply this at-least once guidance to all of the previous one-time-only events such as InstrumentPublished, BeginInstrumentReporting, and EndInstrumentReporting.